### ISSUE TITLE
Search bar loses focus when Escape key is pressed while focused on its options

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-ui-kit#4446]
### Summary of the bug

When you focus on a search bar and then focus on its dropdown menu, pressing Escape causes the focus to be lost completely; it should move back to the input box.

### 🪜 How to reproduce

1. Go to the ICDS website and focus on the search bar.
2. Type in some characters and then press the Down arrow key to move the focus to the dropdown menu.
3. Press the Escape key and see that the focus is lost completely; the focus indicator is not visible at all anywhere.

If I use a screen reader, the focus just moves back to the whole tab, i.e. it says "Home - Intelligence Community Design System".

### 📸 Screenshots or code

https://github.com/user-attachments/assets/8ca25fa0-8bbe-4c99-9f18-603f76058951

### 🧐 Expected behaviour

The focus should move back to the input box (just like what happens with the select component).

### 🚨 Urgency (low, medium or high)

Probably high urgency as it's an accessibility issue
